docs/fsharp/whats-new/fsharp-9.md

```diff
@@ -0,0 +1,432 @@
+
+```fsharp
+[&lt;Fact&gt;]
+let ``this test always fails`` =
+  Assert.True(false)
+```
+
+Run the tests with `dotnet test` and they would pass. Since the test function is not actually a function, it is ignored by the test runner.
+
+Now, with correct attribute enforcement, you will get an `error FS0842: This attribute is not valid for use on this language element`.
+
+## Updates to the standard library (FSharp.Core)
+
+### Random functions for collections
+
+The `List`, `Array`, and `Seq` modules have new functions for random sampling and shuffling. This makes F# easier to use for common data science, machine learning, game development, and other scenarios where randomness is needed.
+
+All functions have variants using:
+
+* An implicit thread-safe shared [`Random`](https://learn.microsoft.com/dotnet/api/system.random) instance
+* Taking a `Random` instance as an argument
+* Taking a custom `randomizer` function, which should return a float value greater than or equal to 0.0 and less than 1.0
+
+There are four functions (each with three variants) available:
+
+#### Shuffle
+
+The shuffle functions return a new collection of the same type and size, with each item in a randomly mixed position. The chance to end up in any position is weighted evenly on the length of the collection.
+
+```fsharp
+let allPlayers = [ "Alice"; "Bob"; "Charlie"; "Dave" ]
+let round1Order = allPlayers |> List.randomShuffle // [ "Charlie"; "Dave"; "Alice"; "Bob" ]
+```
+
+For arrays, there are also `InPlace` variants that shuffle the array in place.
+
+#### Choice
+
+The choice functions return a single random element from the given collection. The random choice is weighted evenly on the size of the collection.
+
+```fsharp
+let allPlayers = [ "Alice"; "Bob"; "Charlie"; "Dave" ]
+let randomPlayer = allPlayers |> List.randomChoice // "Charlie"
+```
+
+#### Choices
+
+Choices select N elements from the input collection in random order, allowing elements to be selected more than once.
+
+```fsharp
+let weather = [ "Raining"; "Sunny"; "Snowing"; "Windy" ]
+let forecastForNext3Days = weather |> List.randomChoices 3 // [ "Windy"; "Snowing"; "Windy" ]
+```
+
+#### Sample
+
+Sample selects N elements from the input collection in random order, without allowing elements to be selected more than once. N cannot be greater than the collection length.
+
+```fsharp
+let foods = [ "Apple"; "Banana"; "Carrot"; "Donut"; "Egg" ]
+let today'sMenu = foods |> List.randomSample 3 // [ "Donut"; "Apple"; "Egg" ]
+```
+
+For a full list of functions and their variants, see ([RFC #1135](https://github.com/fsharp/fslang-design/blob/main/RFCs/FS-1135-random-functions-for-collections.md)).
+
+### Parameterless constructor for `CustomOperationAttribute`
+
+This makes it easier to create a custom operation for a computation expression builder. It will use the name of the method instead of having to explicitly name it (when in most cases the name matches the method name already).
```

Comment:
```suggestion
This constructor makes it easier to create a custom operation for a computation expression builder. It uses the name of the method instead of having to explicitly name it (when in most cases the name matches the method name already).
```